src/component/mapping/new-mapping-destination.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import remote from 'remote';
+import createChooseFile from '../../service/choose-file.js';
+
+const {bool, string, func} = React.PropTypes;
+
+const NewMappingTarget = (props) => {
```

Comment:
This feels like it should be two different components: `FileMappingTarget` and `UrlMappingTarget`. It feels like a lot of if/else all the way down for whether or not it's a `local` mapping.